packages/jest-jasmine2/src/queueRunner.ts

```diff
@@ -27,15 +26,24 @@ export type QueueableFn = {
   initError?: Error;
 };
 
-// har to type :(
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export default function queueRunner(options: Options) {
-  const token = new PCancelable((onCancel: Function, resolve: Function) => {
+type PromiseCallback = (() => void | PromiseLike<void>) | undefined | null;
+
+export default function queueRunner(
+  options: Options,
+): {
+  cancel: () => void;
+  catch: (onRejected?: PromiseCallback) => Promise<void>;
+  then: (
+    onFulfilled?: PromiseCallback,
+    onRejected?: PromiseCallback,
+  ) => Promise<void>;
+} {
```

Comment:
```suggestion
): PCancelable {
```

?

Reply:
That was my initial hope too, but sadly it would require more work than I think probably worth:

```
Error:(89, 3) TS2740: Type '{ cancel: () => void; catch: <TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | null | undefined) => Promise<void | TResult>; then: <TResult1 = void, TResult2 = never>(onfulfilled?: ((value: void) => TResult1 | PromiseLike<...>) | ... 1 more ... | undefined, onrejected?: ((reason: any...' is missing the following properties from type 'PCancelable<void>': _pending, _canceled, _promise, _reject, and 2 more.
```

It's because we're racing against the `PCancelable` promise for all `queueableFns` (which also means cancelling cancels all the promises rather than a specific one), so we return the object `{ cancel: token.cancel, catch: result.catch, then: result.then }` rather than a single Promise from anywhere.

It might be possible to refactor to use the `PCancelable`, but I don't think it's worth it.

We also can't use `Promise<T>` because that requires the `[Symbol.toStringTag]` property, which I'd like to avoid having to implement (not that it's hard per say, just I'd prefer to let JS handle it).

We can however use `PromiseLike<T>`, which would remove the big `then`.

Reply:
aight, fair enough 👍 Thanks for the thorough explanation